src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -265,7 +255,6 @@ private BoundExpression VisitArrayOrSpanCollectionExpression(BoundCollectionExpr
         {
             Debug.Assert(!_inExpressionLambda);
             Debug.Assert(_additionalLocals is { });
-            Debug.Assert(node.CollectionCreation is null); // shouldn't have generated a constructor call
```

Comment:
Why was the assertion removed? Looking at conversion logic, I don't see any changes affecting `collectionCreation` outside of collection builder scenario.

Reply:
`VisitArrayOrSpanCollectionExpression()` is called from `VisitCollectionBuilderCollectionExpression()`, when generating the `ReadOnlySpan<T>` argument for the builder method, even though `node` represents the collection created from the builder in that case, rather than the argument.